CUMULUS-2165 database provisioning/deployment instructions (#2172)
nasa/cumulus: docs/deployment/README.md

- Creating [AWS S3 Buckets](https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingBucket.html)
- Configuring a VPC, if necessary
- Configuring an Earthdata application, if necessary

> - Creating/configuring a Postgres 10.2 compatible database, and an AWS KMS secret to allow database access

Author: No, it should be Secrets Manager.   01756054b

> - Creating/configuring a Postgres 10.2 compatible database, and an AWS KMS secret to allow database access

Reviewer: I think that this particular instance was still not addressed

> - Creating/configuring a Postgres 10.2 compatible database, and an AWS KMS secret to allow database access

Author: Right you are c5abd53cf

---

## Configure the Postgres database

> Cumulus requires a Postgres 10.2 compatible database cluster deployed to AWS.    We suggest utilizing [RDS](https://docs.aws.amazon.com/rds/index.html), and have provided a default [template and RDS cluster module](postgres_database_deployment) utilizing Aurora Serverless. However, Core intentionally provides a "bring your own" approach, and any well-planned cluster setup should work, given the following:

Reviewer: Edit: yes this does work

> Cumulus requires a Postgres 10.2 compatible database cluster deployed to AWS.    We suggest utilizing [RDS](https://docs.aws.amazon.com/rds/index.html), and have provided a default [template and RDS cluster module](postgres_database_deployment) utilizing Aurora Serverless. However, Core intentionally provides a "bring your own" approach, and any well-planned cluster setup should work, given the following:

Reviewer: ```suggestion
Cumulus requires a PostgreSQL 10.2 compatible database cluster deployed to AWS.    We suggest utilizing [RDS](https://docs.aws.amazon.com/rds/index.html), and have provided a default [template and RDS cluster module](postgres_database_deployment) utilizing Aurora Serverless. However, Core intentionally provides a "bring your own" approach, and any well-planned cluster setup should work, given the following:
```

> Cumulus requires a Postgres 10.2 compatible database cluster deployed to AWS.    We suggest utilizing [RDS](https://docs.aws.amazon.com/rds/index.html), and have provided a default [template and RDS cluster module](postgres_database_deployment) utilizing Aurora Serverless, however Core intentonally provides for a "bring your own" approach, and any well-planned cluster setup should work, given the following:

Reviewer 2: ```suggestion
Cumulus requires a Postgres 10.2 compatible database cluster deployed to AWS.    We suggest utilizing [RDS](https://docs.aws.amazon.com/rds/index.html), and have provided a default [template and RDS cluster module](postgres_database_deployment) utilizing Aurora Serverless. However, Core intentionally provides a "bring your own" approach, and any well-planned cluster setup should work, given the following:
```

- Appropriate testing/evaluation is given to ensure the database capacity will scale and the database deployment will allow access to Cumulus's internal components.   Core provides for security-group oriented permissions management via the `rds_security_group` configuration parameter.

> - The database is configured such that it's endpoint is accessible from the VPC and subnets configured for the Core deployment.

Reviewer 2: ```suggestion
- The database is configured such that its endpoint is accessible from the VPC and subnets configured for the Core deployment.
```

> - An AWS Secrets Manager secret exists that the Cumulus deployment user can assign IAM roles with access that has the following format:

Author: Hmm, fair.  That could be better.   

How about: 2501e18f5, fc94071f1

The idea is that whatever secret is created has to be done in such a way that the AWS credentials we're deploying with can assign it to Core lambdas/etc for access.

> - An AWS Secrets Manager secret exists that the Cumulus deployment user can assign IAM roles with access that has the following format:

Reviewer: So right now we have

> An AWS Secrets Manager secret exists that the Cumulus deployment AWS credentials can create/update IAM roles to access that has the following format:

Translation for myself: You need a Secrets manager secret that IAM roles will be created/updated to be able to access. Is that correct?

If so, is the note about updating IAM roles here really necessary? It feels a bit confusing. And it's more of a detail about how the secret is used by our deployment code internally and less about what users need to do the "bring your own database" approach. I'm inclined to just remove the detail here about IAM

> - An AWS Secrets Manager secret exists that the Cumulus deployment user can assign IAM roles with access that has the following format:

Author: > Translation for myself: You need a Secrets manager secret that IAM roles will be created/updated to be able to access. Is that correct?

Not precisely, but looking through a bit more on the particulars of what's possible with SM secret access restriction, I think I agree this is unneeded. 

1e0a8b749

```json
{
  "database": "databaseName",
  "host": "xxx",
  "password": "defaultPassword",
  "port": 5432,
  "username": "xxx"
}
```

> - `database` -- the postgres database used by the configured user

Reviewer: ```suggestion
- `database` -- the PostgreSQL database used by the configured user
```

- `host` -- the RDS service host for the database in the form (dbClusterIdentifier)-(AWS ID string).(region).rds.amazonaws.com
- `password` -- the database password
- `port` -- The database connection port, should always be 5432

> - `username` -- the account username

Reviewer: ```suggestion
- `username` -- the database username
```

It should be the username for the database, correct?

Author: 👍🏻

> This secret should provide access to a postgres database provisioned on the cluster.

Reviewer: ```suggestion
This secret should provide access to a PostgreSQL database provisioned on the cluster.
```
